File: domain/src/main/java/com/study/platform/domain/post/model/StudyPostRepository.java
package com.study.platform.domain.post.model;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface StudyPostRepository {

    StudyPost save(StudyPost post);
    List<StudyPost> saveAll(List<StudyPost> posts);
    void delete(StudyPost post);
    Optional<StudyPost> findById(UUID id);
    Optional<StudyPost> findByIdWithAuthor(UUID postId);
    List<StudyPost> findExpiredPosts(LocalDateTime now, StudyPostStatus status);
    List<StudyPost> findDeadlineReminderPosts(LocalDateTime start, LocalDateTime end, StudyPostStatus status);
    Optional<StudyPost> findByIdWithAuthorForUpdate(UUID postId);
}

File: domain/src/main/kotlin/com/study/platform/domain/post/model/StudyPostRepository.kt
package com.study.platform.domain.post.model

import java.time.LocalDateTime
import java.util.UUID

interface StudyPostRepository {

    fun save(post: StudyPost): StudyPost
    fun saveAll(posts: List<StudyPost>): List<StudyPost>
    fun delete(post: StudyPost)
    fun findById(id: UUID): StudyPost?
    fun findByIdWithAuthor(postId: UUID): StudyPost?
    fun findExpiredPosts(now: LocalDateTime, status: StudyPostStatus): List<StudyPost>
    fun findDeadlineReminderPosts(start: LocalDateTime, end: LocalDateTime, status: StudyPostStatus): List<StudyPost>
    fun findByIdWithAuthorForUpdate(postId: UUID): StudyPost?
}
